modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5026,7 +5026,12 @@ void GDScriptAnalyzer::is_shadowing(GDScriptParser::IdentifierNode *p_identifier
 			parser->push_warning(p_identifier, GDScriptWarning::SHADOWED_GLOBAL_IDENTIFIER, p_context, name, "built-in function");
 			return;
 		} else if (ClassDB::class_exists(name)) {
-			parser->push_warning(p_identifier, GDScriptWarning::SHADOWED_GLOBAL_IDENTIFIER, p_context, name, "global class");
+			parser->push_warning(p_identifier, GDScriptWarning::SHADOWED_GLOBAL_IDENTIFIER, p_context, name, "native class");
+			return;
+		} else if (ScriptServer::is_global_class(name)) {
+			String class_path = ScriptServer::get_global_class_path(name);
+			class_path = class_path.substr(class_path.rfind("/") + 1);
```

Comment:
```suggestion
			String class_path = ScriptServer::get_global_class_path(name).get_file();
```

There is [`get_file()`](https://docs.godotengine.org/en/stable/classes/class_string.html#class-string-method-get-file) method. I'm not sure if it's worth truncating the path, but unfortunately the tests now use absolute paths, so the choice is between the file name and not using the path at all.